src/main/java/de/cas/mse/exercise/csv/CsvReader.java

```diff
@@ -1,19 +1,89 @@
 package de.cas.mse.exercise.csv;
 
+import java.io.BufferedReader;
 import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
 
 import de.cas.mse.exercise.csv.ui.CsvUi;
+import de.cas.mse.exercise.csv.ui.Filterable;
 
-public class CsvReader {
-	
-	private CsvUi csvUi;
+public class CsvReader implements Filterable {
+    
+    private static final String CSV_DELIMETER = ",";
+    private static final int VIEW_TO_MODEL_COLUMN_OFFSET = 1;
+
+    private CsvUi csvUi;
+
+    private CsvData csvData;
 
 	public void run(final File csvFile) throws Exception {
-		// TODO: Start coding here
+        csvData = generateCsvDataFromFile(csvFile);
+        initializeCsvUi(csvData);
 	}
 	
 	public void setCsvUi(final CsvUi csvUi) {
 		this.csvUi = csvUi;
-	}
+    }
+
+    @Override
+    public List<List<String>> filter(int viewCol, String filterText) {
+        return csvData.getRows().stream() //
+            .filter(row -> row.get(viewToModelColumnIndex(viewCol)).contains(filterText)) //
+            .collect(Collectors.toList());
+    }
+
+    private int viewToModelColumnIndex(int viewColumnIndex) {
+        return viewColumnIndex - VIEW_TO_MODEL_COLUMN_OFFSET;
+    }
+
+    private CsvData generateCsvDataFromFile(File csvFile) throws IOException {
+        CsvData csvData = null;
+        try (BufferedReader reader = new BufferedReader(new FileReader(csvFile))) {
+            List<String> headerRow = parseCsvLine(reader.readLine());
+            csvData = new CsvData(headerRow);
+            String csvRow;
+            while ((csvRow = reader.readLine()) != null) {
+                csvData.addRow(parseCsvLine(csvRow));
```

Comment:
maybe this interface can be improved by providing the complete data set at once by using streams